src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-list/rbd-snapshot-actions.model.ts

```diff
@@ -5,54 +5,63 @@ export class RbdSnapshotActionsModel {
   create: CdTableAction = {
     permission: 'create',
     icon: 'fa-plus',
-    buttonCondition: (selection: CdTableSelection) => !selection.hasSingleSelection,
+    buttonCondition: (selection: CdTableSelection) =>
+      !selection.hasSingleSelection,
     name: 'Create'
   };
   rename: CdTableAction = {
     permission: 'update',
     icon: 'fa-pencil',
+    disable: (selection: CdTableSelection) =>
+      !selection.hasSingleSelection || selection.first().cdExecuting,
     name: 'Rename'
   };
   protect: CdTableAction = {
     permission: 'update',
     icon: 'fa-lock',
     visible: (selection: CdTableSelection) =>
       selection.hasSingleSelection && !selection.first().is_protected,
+    disable: (selection: CdTableSelection) =>
+      !selection.hasSingleSelection || selection.first().cdExecuting,
     name: 'Protect'
   };
   unprotect: CdTableAction = {
     permission: 'update',
     icon: 'fa-unlock',
     visible: (selection: CdTableSelection) =>
       selection.hasSingleSelection && selection.first().is_protected,
+    disable: (selection: CdTableSelection) =>
```

Comment:
This is not needed. Table Actions already takes care of disabling "update" and "delete" actions.